pkg/handler/v2/provider/kubevirt_test.go

```diff
@@ -0,0 +1,553 @@
+	return &kvapiv1.VirtualMachineInstancePresetSpec{
+		Domain: &kvapiv1.DomainSpec{
+			CPU: &kvapiv1.CPU{
+				Cores: 2,
+			},
+			Devices: kvapiv1.Devices{
+				Disks: []kvapiv1.Disk{
+					{
+						Name:       "datavolumedisk",
+						DiskDevice: kvapiv1.DiskDevice{Disk: &kvapiv1.DiskTarget{Bus: "virtio"}},
+					},
+					{
+						Name:       "cloudinitdisk",
+						DiskDevice: kvapiv1.DiskDevice{Disk: &kvapiv1.DiskTarget{Bus: "virtio"}},
+					},
+				},
+			},
+			Resources: kvapiv1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceMemory: *resource.NewQuantity(memoryReq, resource.Format("BinarySI")),
+					corev1.ResourceCPU:    *resource.NewQuantity(cpuReq, resource.Format("BinarySI")),
+				},
+				Limits: corev1.ResourceList{
+					corev1.ResourceMemory: *resource.NewQuantity(memoryLimit, resource.Format("BinarySI")),
+					corev1.ResourceCPU:    *resource.NewQuantity(cpuLimit, resource.Format("BinarySI")),
+				},
+			},
+		}}
+}
+
+func NewCredentialSecret(name, namespace string) *corev1.Secret {
+	data := map[string][]byte{
+		"kubeConfig": []byte(fakeKvConfig),
+	}
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Data: data,
+	}
+}
+
+func GenKubevirtKubermaticPreset() *kubermaticv1.Preset {
+	return &kubermaticv1.Preset{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "kubermatic-preset",
+		},
+		Spec: kubermaticv1.PresetSpec{
+			Kubevirt: &kubermaticv1.Kubevirt{
+				Kubeconfig: fakeKvConfig,
+			},
+			Fake: &kubermaticv1.Fake{Token: "dummy_pluton_token"},
+		},
+	}
+}
+
+var (
+	presetListResponse = `[{"name":"preset-default-small-1","creationTimestamp":"0001-01-01T00:00:00Z","spec":{"selector":{},"domain":{"resources":{"limits":{"cpu":"456","memory":"345"},"requests":{"cpu":"234","memory":"123"}},"cpu":{"cores":2},"devices":{"disks":[{"name":"datavolumedisk","disk":{"bus":"virtio"}},{"name":"cloudinitdisk","disk":{"bus":"virtio"}}]}}}},{"name":"preset-default-small-2","creationTimestamp":"0001-01-01T00:00:00Z","spec":{"selector":{},"domain":{"resources":{"limits":{"cpu":"456","memory":"345"},"requests":{"cpu":"234","memory":"123"}},"cpu":{"cores":2},"devices":{"disks":[{"name":"datavolumedisk","disk":{"bus":"virtio"}},{"name":"cloudinitdisk","disk":{"bus":"virtio"}}]}}}}]`
+)
+
+var (
+	reclaimPolicy = v1.PersistentVolumeReclaimDelete
+	storageClass1 = storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "storageclass-1",
+		},
+		ReclaimPolicy: &reclaimPolicy,
+	}
+	storageClass2 = storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "storageclass-2",
+		},
+	}
+
+	storageClassListResponse = `{"metadata":{},"items":[{"metadata":{"name":"storageclass-1","creationTimestamp":null},"provisioner":"","reclaimPolicy":"Delete"},{"metadata":{"name":"storageclass-2","creationTimestamp":null},"provisioner":""}]}`
+)
+
+func TestEndpoint(t *testing.T) {
```

Comment:
It's really hard to read the test scenarios. Can you split the tests for different endpoints like for other providers:
 - api/v2/providers/kubevirt/vmipresets
 - /api/v2/projects/%s/clusters/%s/providers/kubevirt/vmipresets

Reply:
I've re-pushed with the different endpoints splitted.